src/libraries/System.Security.Cryptography/tests/OpenSslNamedKeysTests.manual.cs

```diff
@@ -536,5 +550,154 @@ public static void Provider_TPM2DecryptRsa_ExportParameters()
             Assert.NotEqual(encrypted, data);
             Assert.Equal(data, rsaPri.Decrypt(encrypted, padding));
         }
+
+        [ConditionalFact(nameof(ShouldRunProviderEcDsaTests))]
+        public static void Provider_TPM2SslStream_ServerCertIsTpmEcDsa()
+        {
+            using X509Certificate2 serverCert = CreateSelfSignedEcDsaCertificate();
+            CreateDefaultTlsOptions(serverCert, out SslServerAuthenticationOptions serverOptions, out SslClientAuthenticationOptions clientOptions);
+            TestTls(serverOptions, clientOptions);
+        }
+
+        [ConditionalTheory(nameof(ShouldRunProviderRsaTests))]
+        [MemberData(nameof(SignaturePaddingValues))]
+        public static void Provider_TPM2SslStream_ServerCertIsTpmRsa(RSASignaturePadding padding)
+        {
+            using X509Certificate2 serverCert = CreateSelfSignedRsaCertificate(padding);
+            CreateTlsOptionsForRsa(serverCert, out SslServerAuthenticationOptions serverOptions, out SslClientAuthenticationOptions clientOptions);
+            TestTls(serverOptions, clientOptions);
+        }
+
+        private static void TestTls(SslServerAuthenticationOptions serverOptions, SslClientAuthenticationOptions clientOptions)
```

Comment:
IMO: TLS tests belong in system.net.security.  Rather than you sharing their tests down to crypto, you should share whatever you need for manual key-handle tests up to them.